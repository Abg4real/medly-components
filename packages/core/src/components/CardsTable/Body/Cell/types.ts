import { ColumnConfig } from '../../types';

export interface Props {
    data: any;
    column: ColumnConfig;
    rowId: any;
    isRowClickDisabled?: boolean;
}

export interface StyledProps {
    flex: number;
    align: 'left' | 'center' | 'right';
}
