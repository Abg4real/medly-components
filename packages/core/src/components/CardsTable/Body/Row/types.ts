import { WithThemeProp } from '@medly-components/utils';

export interface Props extends WithThemeProp {
    onClick?: () => void;
    withWhiteBackground: boolean;
}
