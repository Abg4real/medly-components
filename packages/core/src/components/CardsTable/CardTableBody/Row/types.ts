import { Props as CardProps } from '../../../Card/types';

export interface Props extends CardProps {
    disabled?: boolean;
    withLightTheme: boolean;
}
