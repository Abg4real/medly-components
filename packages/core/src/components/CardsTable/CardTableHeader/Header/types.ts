import { SortOrder } from '../../types';
import { WithThemeProp } from '@medly-components/utils/src';

export interface Props extends WithThemeProp{
    sortField?: string;
    field?: string;
    onSortChange?: (field: string, order: SortOrder) => void;
    enableSorting?: boolean;
    title: string;
}    