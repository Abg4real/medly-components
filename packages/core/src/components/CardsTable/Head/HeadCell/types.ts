import { WithThemeProp } from '@medly-components/utils/src';
import { ColumnConfig, SortOrder } from '../../types';

export interface Props extends WithThemeProp {
    column: ColumnConfig;
    sortField?: string;
    onSortChange?: (field: string, order: SortOrder) => void;
}

export interface StyledProps {
    flex: number;
    align: 'left' | 'center' | 'right';
}
