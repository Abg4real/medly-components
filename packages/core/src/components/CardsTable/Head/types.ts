import { ColumnConfig, SortOrder } from '../types';

export interface Props {
    columns: ColumnConfig[];
    onSort?: (field: string, order: SortOrder) => void;
}
