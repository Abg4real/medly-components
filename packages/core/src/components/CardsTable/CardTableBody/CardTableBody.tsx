import { WithStyle } from '@medly-components/utils';
import React, { SFC } from "react";
import * as Style from './CardTableBody.styled';
import Cell from './Cell';
import Row from './Row';
import { Props } from "./types";


export const CardTableBody: SFC<Props> & WithStyle= React.memo( props => {
    const { data, columns, onRowClick, rowClickDisableKey, uniqueKeyName, withLightTheme} = props;
    
    const handleRowClick = (rowData: any) => {
        return onRowClick && !rowData[rowClickDisableKey] ? () => onRowClick(rowData) : undefined;
    };

    return (
        <>
            {data.map((row, index) => {
                return (
                    <Row
                        key={index} 
                        flowDirection="horizontal" 
                        fullWidth 
                        withPadding={false} 
                        onClick={handleRowClick(row)}
                        disabled={row[rowClickDisableKey]}
                        withLightTheme={withLightTheme}
                        >
                        { columns.map(column => (
                            <Style.Column key={`${column.field}-${index}`} variant="flat" flowDirection="vertical" alignItems="left">
                               <Cell 
                                    isRowClickDisabled={row[rowClickDisableKey]}
                                    data={row[column.field]}
                                    column={column}
                                    rowId={row[uniqueKeyName]}
                               />
                            </Style.Column>
                        )) }
                    </Row>
                )})
            } 
        </>         
    )  
});

CardTableBody.displayName = 'CardTableBody';
