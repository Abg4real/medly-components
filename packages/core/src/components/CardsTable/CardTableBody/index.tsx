import { CardTableBody } from './CardTableBody';

export default CardTableBody;