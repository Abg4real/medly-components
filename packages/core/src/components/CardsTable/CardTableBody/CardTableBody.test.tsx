import { render } from '@test-utils';
import React from 'react';
import { CardTableBody } from './CardTableBody'
import { ColumnConfig } from '../types';

describe('CardTableBody', () => {
    it('should render cart table body properly', () => {

        const data = [{
            id: 1,
            patientInfo: 'Oli Bob',
            rxInfo: 'Metaformin',
            price: '535.76',
            status: 'Pending'
            
        }];
        const columns: ColumnConfig[] = [
            { title: 'Patient Info', field: 'patientInfo', formatter: 'text-short', sort: true },
            { title: 'RxInfo', field: 'rxInfo', formatter: 'group', sort: true }
        ];

        const onRowClick = jest.fn();
        
        const { container } = render(<CardTableBody 
            data= {data}
            columns={columns}
            onRowClick={onRowClick}
            rowClickDisableKey="disable"
            uniqueKeyName="124"

         />);
        expect(container).toMatchSnapshot();
    });
});
