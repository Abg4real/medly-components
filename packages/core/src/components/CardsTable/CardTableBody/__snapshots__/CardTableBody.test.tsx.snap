// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardTableBody should render cart table body properly 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  margin: 5px;
  overflow: auto;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  box-sizing: border-box;
  background-color: #ffffff;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 4px;
  border: 1px solid #dfe4e9;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.2);
  width: calc(100% - 10px);
}

.c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  margin: 5px;
  overflow: auto;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  box-sizing: border-box;
  background-color: #ffffff;
  padding: 10px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  border-radius: 4px;
  border: 1px solid #dfe4e9;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.2);
  border-radius: 0;
  margin: 0;
  border: none;
  box-shadow: none;
  background-color: transparent;
}

.c2.c2 * {
  background-color: transparent;
}

.c3 {
  padding: 15px 20px;
}

.c4 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
}

.c1 {
  cursor: -webkit-grab;
  cursor: -moz-grab;
  cursor: grab;
  box-shadow: none;
  box-sizing: border-box;
  border-radius: 0.8rem;
  border: none;
}

.c1:hover {
  box-shadow: 0px 4px 8px #b0bcc8;
}

.c1 > * {
  background-color: inherit;
}

.c1:nth-child(2n) {
  background: #ffffff;
}

<div>
  <div
    class="c0 c1"
  >
    <div
      class="c2 c3"
    >
      <span
        class="c4"
      >
        Oli Bob
      </span>
    </div>
    <div
      class="c2 c3"
    >
      <span
        class="c4"
      >
        Metaformin
      </span>
    </div>
  </div>
</div>
`;
